cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -170,6 +193,99 @@ protected void addPartition(PartitionKey key, String path, boolean explorable, M
   }
 
   @Override
+  public PartitionConsumerResult consumePartitions(PartitionConsumerState partitionConsumerState) {
+    long[] previousInProgress = partitionConsumerState.getVersionsToCheck();
+    long[] inProgress = tx.getInProgress();
+    Long[] noLongerInProgress = setDiff(previousInProgress, inProgress);
+
+    Iterator<Partition> noLongerInProgressPartitions = Iterators.emptyIterator();
+    if (noLongerInProgress.length > 0) {
```

Comment:
it may be better to do point lookups for each of the "no longer in progress". The index scan is relatively expensive because it scans everything in between, and fetches the primary row for everything in between.